dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+        });
+
+        return Flux.usingWhen(Mono.fromSupplier(DotReactiveElasticsearchClient.ScrollState::new),
+
+                scrollState -> {
+
+                    Flux<SearchHit> searchHits = inbound.<SearchResponse> handle((searchResponse, sink) -> {
+
+                        scrollState.updateScrollId(searchResponse.getScrollId());
+                        if (isEmpty(searchResponse.getHits())) {
+
+                            inbound.onComplete();
+                            outbound.onComplete();
+
+                        } else {
+
+                            sink.next(searchResponse);
+
+                            SearchScrollRequest searchScrollRequest = new SearchScrollRequest(scrollState.getScrollId())
+                                    .scroll(scrollTimeout);
+                            request.next(searchScrollRequest);
+                        }
+
+                    }).map(SearchResponse::getHits) //
+                            .flatMap(Flux::fromIterable);
+
+                    return searchHits.doOnSubscribe(ignore -> exchange.subscribe(inbound));
+
+                }, state -> cleanupScroll(headers, state), //
+                state -> cleanupScroll(headers, state), //
+                state -> cleanupScroll(headers, state)); //
+    }
+
+    private static boolean isEmpty(@Nullable SearchHits hits) {
+        return hits != null && hits.getHits() != null && hits.getHits().length == 0;
+    }
+
+    private Publisher<?> cleanupScroll(HttpHeaders headers, DotReactiveElasticsearchClient.ScrollState state) {
+
+        if (state.getScrollIds().isEmpty()) {
+            return Mono.empty();
+        }
+
+        ClearScrollRequest clearScrollRequest = new ClearScrollRequest();
+        clearScrollRequest.scrollIds(state.getScrollIds());
+
+        // just send the request, resources get cleaned up anyways after scrollTimeout has been reached.
+        return sendRequest(clearScrollRequest, DotReactiveElasticsearchClient.RequestCreator.clearScroll(), ClearScrollResponse.class, headers);
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.index.reindex.DeleteByQueryRequest)
+     */
+    public Mono<BulkByScrollResponse> deleteBy(HttpHeaders headers, DeleteByQueryRequest deleteRequest) {
+
+        return sendRequest(deleteRequest, DotReactiveElasticsearchClient.RequestCreator.deleteByQuery(), BulkByScrollResponse.class, headers) //
+                .publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#bulk(org.springframework.http.HttpHeaders, org.elasticsearch.action.bulk.BulkRequest)
+     */
+    @Override
+    public Mono<BulkResponse> bulk(HttpHeaders headers, BulkRequest bulkRequest) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'headers' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)